### DISCUSSION TITLE
Is there a way to pretty print a tensor ?

### DISCUSSION THREAD

**Discussion by sreeramcoder** [themetaschemer/malt#43] [Q&A] [answered]
Hello,

For the sample input 

`(display (+ (tensor 2) (tensor 7))) 
`
The output is as below.

`#(#(#<procedure:dual> 9 #<procedure:...utodiff/B-prims.rkt:20:10>)) `

Is there a function available in the library to pretty print the tensor ?

For nested tensors, it might be helpful to just print the values.

Thank you,
Sreeram

**Comment by themetaschemer** [comment]
Hi Sreeram, sorry I missed this question of yours. When you load malt, it should automatically adjust the printing ports so that the output of tensors is properly printed. Are you printing to another port? See, for example, the output that I get: 

```
(require malt)
> (display (+ (tensor 2) (tensor 7)))
(display (+ (tensor 2) (tensor 7)))
(tensor 9)
```

**Comment by sreeramcoder** [comment]
Hi,

I am seeing that this error occurs only when using malt/learner and after using a extended function.

For e.g., below is a minimal example to replicate the issue.

```
(base) sreeram@localhost ~/prog/ll $ cat test.rkt
#lang racket

(require malt/learner)

(define T (tensor 1 2 3))

(display (sqr T))

(base) sreeram@localhost ~/prog/ll $ racket test.rkt
#(#(#<procedure:dual> 1 #<procedure:...utodiff/B-prims.rkt:20:10>) #(#<procedure:dual> 4 #<procedure:...utodiff/B-prims.rkt:20:10>) #(#<procedure:dual> 9 #<procedure:...utodiff/B-prims.rkt:20:10>))

````

Below works fine.


```
(base) sreeram@localhost ~/prog/ll $ cat test-2.rkt 
#lang racket

(require malt)

(define T (tensor 1 2 3))

(display (sqr T))

(base) sreeram@localhost ~/prog/ll $ racket test-2.rkt 
(tensor 1 4 9)

```

**Reply by themetaschemer** [reply]
Yes, that might be the issue. Let me investigate.

**Comment by themetaschemer** [comment, marked as answer]
I will push a fix for this, but in the meantime, here's a suggested workaround:

```
(require malt/learner)
(require malt/ports)
(printable-maker make-printable)

```
This should give you the following: 
```
> (+ (tensor 1 2 3) (tensor 4 5 7))
(tensor 5 7 10)

```

**Reply by sreeramcoder** [reply]
Thank you !
